gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/gluten/uniffle/GlutenRssShuffleManager.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.uniffle.common.exception.RssException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+
+public class GlutenRssShuffleManager extends RssShuffleManager {
+  private static final Logger LOG = LoggerFactory.getLogger(GlutenRssShuffleManager.class);
+  private static final String GLUTEN_SHUFFLE_MANAGER_NAME =
+      "org.apache.spark.shuffle.sort.ColumnarShuffleManager";
+
+  private static final String VANILLA_UNIFFLE_SHUFFLE_MANAGER_NAME =
+      "org.apache.spark.shuffle.celeborn.SparkShuffleManager";
+  private volatile ColumnarShuffleManager _columnarShuffleManager;
+  private volatile RssShuffleManager _vanillaUniffleShuffleManager;
+
+  private ColumnarShuffleManager columnarShuffleManager() {
+    if (_columnarShuffleManager == null) {
+      synchronized (this) {
+        if (_columnarShuffleManager == null) {
+          _columnarShuffleManager =
+              initShuffleManager(GLUTEN_SHUFFLE_MANAGER_NAME, sparkConf, isDriver());
+        }
+      }
+    }
+    return _columnarShuffleManager;
+  }
+
+  private RssShuffleManager vanillaUniffleShuffleManager() {
+    if (_vanillaUniffleShuffleManager == null) {
+      synchronized (this) {
+        if (_vanillaUniffleShuffleManager == null) {
+          initShuffleManager(VANILLA_UNIFFLE_SHUFFLE_MANAGER_NAME, sparkConf, isDriver());
+        }
+      }
+    }
+    return _vanillaUniffleShuffleManager;
+  }
+
+  private boolean isDriver() {
+    return "driver".equals(SparkEnv.get().executorId());
+  }
+
+  private ColumnarShuffleManager initShuffleManager(String name, SparkConf conf, boolean isDriver) {
+    Constructor constructor;
+    ColumnarShuffleManager instance;
+    try {
+      Class klass = Class.forName(name);
+      try {
+        constructor = klass.getConstructor(conf.getClass(), Boolean.TYPE);
+        instance = (ColumnarShuffleManager) constructor.newInstance(conf, isDriver);
+      } catch (NoSuchMethodException var7) {
+        constructor = klass.getConstructor(conf.getClass());
+        instance = (ColumnarShuffleManager) constructor.newInstance(conf);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("initColumnManager fail");
+    }
+    return instance;
+  }
+
+  public GlutenRssShuffleManager(SparkConf conf, boolean isDriver) {
+    super(conf, isDriver);
+    conf.set("spark.rss.row.based", "false");
+  }
+
+  @Override
+  public <K, V, C> ShuffleHandle registerShuffle(
+      int shuffleId, ShuffleDependency<K, V, C> dependency) {
+    return super.registerShuffle(shuffleId, dependency);
+  }
+
+  @Override
+  public <K, V> ShuffleWriter<K, V> getWriter(
+      ShuffleHandle handle, long mapId, TaskContext context, ShuffleWriteMetricsReporter metrics) {
+    if (!(handle instanceof RssShuffleHandle)) {
+      throw new RssException("Unexpected ShuffleHandle:" + handle.getClass().getName());
+    }
+    sparkConf.setIfMissing(
```

Comment:
Do we need this code? Seems the setting during the instantiation of `GlutenRssShuffleManager` is enough.